### PR TITLE
Add filesize w/ git auto-update

### LINKED ISSUE #10970: [Request] Add filesize.js
**Library name:** filesize.js
**Git repository url:** https://github.com/avoidwork/filesize.js
**npm package name or url** (if there is one): https://www.npmjs.com/package/filesize
**License (List them all if it's multiple):** BSD 3-clause
**Official homepage:** https://filesizejs.com/
**Wanna say something? Leave message here:** JavaScript library to generate a human readable String describing the file size

=====================

Notes from cdnjs maintainer:
Please read the [README.md](https://github.com/cdnjs/cdnjs#cdnjs-library-repository) and [CONTRIBUTING.md](https://github.com/cdnjs/cdnjs/blob/master/.github/CONTRIBUTING.md) document first.

We encourage you to add a library via sending pull request,
it'll be faster than just opening a request issue,
since there are tons of issues, please wait with patience,
and please don't forget to read the guidelines for contributing, thanks!!

### PR DESCRIPTION
**Pull request by dakshshah96**
close #10970, cc @avoidwork

Pull request for issue: #10970 

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/avoidwork/filesize.js
 * Official website (optional, not the repository): https://filesizejs.com/
 * NPM package url (optional): https://www.npmjs.com/package/filesize
 * License and its reference: BSD-3-Clause
 * GitHub / Bitbucket popularity (required): 
   - Count of watchers: 13
   - Count of stars: 385
   - Count of forks: 41
 * NPM download stats (optional):
   - Downloads in the last day: 65,619
   - Downloads in the last week: 365,212
   - Downloads in the last month: 1,450,513

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/filesize/3.5.6/filesize.js [hunk 1]
```diff
@@ -0,0 +1,167 @@
+"use strict";
```
Comment (hare1039) [inline ajax/libs/filesize/3.5.6/filesize.js]: Although the content looks same, but my diff & md5sum reports difference
files I download from npm:
```
6c75124acbf55786da221c4b03353daf  ./3.5.6/filesize.es6.js
5edc9424204684ed6ad0777093abd5fb  ./3.5.6/filesize.js
```
files in this PR:
```
7cb7dc343b941547fc1e28c7d93ba000  ./3.5.6/filesize.es6.js
f7468628e89bd7a83c183ade8b8bbaea  ./3.5.6/filesize.js
```

Can you check these file again. Lots of thanks!

File: ajax/libs/filesize/package.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    "file system",
+    "bytes",
+    "diff"
+  ],
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/avoidwork/filesize.js"
+  },
+  "license": "BSD-3-Clause",
+  "author": "Jason Mulligan <@avoidwork> (https://avoidwork.com/)",
```
Comment (avoidwork) [inline ajax/libs/filesize/package.json]: The form of that doesn't reflect the actual `package.json`, it'd be `"author": "Jason Mulligan <jason.mulligan@avoidwork.com"`; the bower file content is probably questionable at this time (created & forgotten).

**Comment by dakshshah96** [comment]
Done 👍 

@hare1039

* Change `lib` file rule to `filesize*`
* Update files to match GitHub release v3.5.6

@sufuf3 

* Use git auto-update since there are more versions than npm
* Update `author` field to match upstream package.json (as mentioned by @avoidwork)

**PENDING:** Add old versions in another commit

File: ajax/libs/filesize/package.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    "filesize",
+    "size",
+    "readable",
+    "file system",
+    "bytes",
+    "diff"
+  ],
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/avoidwork/filesize.js"
```
Comment (sufuf3) [inline ajax/libs/filesize/package.json]: Maybe we can follow the format as the same as upstream. ref: https://github.com/avoidwork/filesize.js/blob/master/package.json#L9
`"url": "git://github.com/avoidwork/filesize.js.git"`

**Comment by dakshshah96** [comment]
@sufuf3 @extend1994 I've added the required old versions of the library in a different commit 👍

**Comment by dakshshah96** [comment]
@sufuf3 Done 👍

**Comment by sufuf3** [comment]
@dakshshah96 Great!
And if you could wrap the second commit message to about 72 characters or so, it will be better.
It will be useful to us to maintain the commit logs of the library.

you could refer to the link (https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project), the quality commit message makes using and collaborating with Git a lot easier. 

The second commit message example as the followings:
```
Add old versions of filesize manually, cc #10970

Manually add v1.0.0~v1.6.0, v1.6.5 from GitHub 
and v1.6.1~v1.6.4 from npm due to different file structure
```

Thanks for your contribution.

**Comment by dakshshah96** [comment]
@sufuf3 I did `git commit --amend` which opened the last commit message, set `wrap_width: 72` in my Sublime Text settings, saved & closed the editor after which I did `git push -u origin add/filesize.js -f`.

Is the commit message fine now?

**Comment by pvnr0082t** [comment]
@extend1994 I think the main reason is "git auto-update can't get the files"

**Comment by dakshshah96** [comment]
@extend1994 @pvnr0082t Done 👍

**Comment by pvnr0082t** [comment]
@dakshshah96 Do we really need `filesize.es6.js`? Could you please confirm this? Thanks.

**Comment by avoidwork** [comment]
@pvnr0082t why wouldn't you want it? if client constraints are es6+ why would you want the babel generated crap?

**Comment by dakshshah96** [comment]
@pvnr0082t As @avoidwork rightly said, I think `filesize.es6.js` is required since it is the ES6 version of the filesize library.

**Comment by sufuf3** [comment]
@dakshshah96 would you like to minify the `filesize.es6.js` in the first commit? Because it will be fast when people browsing.
Sorry about that I should remind you at first.

**Comment by dakshshah96** [comment]
@sufuf3 Done 👍

**Comment by sufuf3** [comment]
@dakshshah96 Great.
And I think maybe we can host `filesize.es6.min.js.map` as the same file structure as `filesize.min.js.map`.

Are you use this minify tool --> https://github.com/PeterDaveHello/web-minify-helper ?
If yes, you can go to the target directory and use `path_of_script/minify.sh`. This will generate the `filesize.es6.min.js.map`.

Thank you very much.

**Comment by avoidwork** [comment]
@sufuf3 @dakshshah96 this should be done in the grunt file of the project and not external. You're looking for either an override or collision by doing it this way.

**Comment by avoidwork** [comment]
@sufuf3 @dakshshah96 es6 minified files are in 3.5.7

**Comment by dakshshah96** [comment]
@sufuf3 Updated the PR, please check 👍 
Thanks @avoidwork!

**Comment by sufuf3** [comment]
@avoidwork Thank a lot.
@dakshshah96 Great! 💯

**Comment by dakshshah96** [comment]
@pvnr0082t Didn't get how to make git auto-update work. I ran `./autoupdate.js filesize` in root directory of autoupdate but got the following output:

```
module.js:442
    throw err;
    ^

Error: Cannot find module 'lodash'
    at Function.Module._resolveFilename (module.js:440:15)
    at Function.Module._load (module.js:388:25)
    at Module.require (module.js:468:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/dakshshah/Documents/Work/Git-Works/autoupdate/autoupdate.js:3:9)
    at Module._compile (module.js:541:32)
    at Object.Module._extensions..js (module.js:550:10)
    at Module.load (module.js:458:32)
    at tryModuleLoad (module.js:417:12)
    at Function.Module._load (module.js:409:3)
```

**Comment by pvnr0082t** [comment]
@dakshshah96 `Error: Cannot find module 'lodash'`. Try to run `npm install` to install necessary dependencies.

**Comment by dakshshah96** [comment]
@pvnr0082t Updated files from latest GitHub release v3.5.9.
Please check and confirm 👍

File: ajax/libs/filesize/1.0.0/filesize.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * filesize.js
+ *
+ * @author Jason Mulligan <jason.mulligan@avoidwork.com>
+ * @version 1.0
+ */
+(function(a){"use strict";var b=function(a,b){var c=String(a).indexOf(".")>-1?parseFloat(a):parseInt(a),d=[{kB:0},{MB:1024},{GB:1048576},{TB:1073741824}],e=d.length,f="",g,h,i,j;b=typeof b=="undefined"?2:parseInt(b);while(e--){j=d[e];for(i in j)j.hasOwnProperty(i)&&(g=j[i],h=i);if(c>=g){f=(h==="kB"?c:(c/g).toFixed(b))+h;break}}return f};typeof define=="function"?define("filesize",[],function(){return b}):a.filesize=b})(window)
```
Comment (PeterDaveHello) [inline ajax/libs/filesize/1.0.0/filesize.js]: This is already minified, doesn't need to be minified again, same as some other versions.

**Comment by dakshshah96** [comment]
@PeterDaveHello Done 👍

**Comment by PeterDaveHello** [comment]
Please also mention the related issue number in the last two commits' commit message next time.

**Pull request merged**